### ISSUE TITLE
Unstable loss when running finetune_qwen. py

### ISSUE THREAD

**Issue by Quinn777** [openreasoner/openr#34]
### System Info

python3.8， torch=2.1.2, cuda=11.8, 16* nvidia v100 gpu

### Who can help?

@Gebro13@mengfn, @gzqaq

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the codebase (such as scrips/, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

This is my input parameters:
parser.add_argument("--base_model_path", type=str, default="./Qwen2.5-Math-7B-Instruct/")
parser.add_argument("--model_path", type=str, default="./math-psa-7b/")

parser.add_argument("--per_device_train_batch_size", type=int, default=1)
parser.add_argument("--per_device_eval_batch_size", type=int, default=1)
parser.add_argument("--total_batch_size", type=int, default=256)
parser.add_argument("--local-rank", type=int, default=-1, help="Local rank for distributed training")
parser.add_argument("--gradient_accumulation_steps", type=int, default=16)

parser.add_argument("--learning_rate", type=float, default=1e-4)
parser.add_argument("--datasets", type=str, default="caption_cot_1028")
parser.add_argument("--server", type=str, default='1')

model = AutoModelForCausalLM.from_pretrained(
    model_path,
    # load_in_8bit=True,   # Enables 8-bit quantization
    # device_map="auto",   # Automatically assigns the model to available GPUs/CPUs
    torch_dtype=torch.float16,  # Mixed precision for faster inference
    # torch_dtype=torch.bfloat16,
    # attn_implementation="flash_attention_2",
)

print(model)
lora_config = LoraConfig(
    task_type=TaskType.CAUSAL_LM,  # LoRA for causal language modeling task
    r=8,  # Rank of LoRA
    lora_alpha=32,  # Alpha scaling factor for LoRA
    lora_dropout=0.1,  # Dropout rate for LoRA layers
    target_modules=["q_proj", "v_proj"],  # Apply LoRA to specific layers
)
model = get_peft_model(model, lora_config)
print(model)
model.print_trainable_parameters()
# model.to('cuda:0')
print(model.device)

DATA_PATH = {
    # "train": 'multi-step.json',
    # 'train': 'test.json',
    # "test": os.path.join(args.data_path, 'prm800k_test.json'),
    "train": os.path.join(args.data_path, "math_aps.json"),
    # "train": "../../datasets/processed_data/prm800k/data/phase2_train_new.jsonl",
    # "test": "../../datasets/prm800k-main/prm800k/data/phase2_test_new.jsonl",

}
dataset = load_dataset('json', data_files=DATA_PATH, cache_dir=args.cache_dir)
training_args = TrainingArguments(
    output_dir=output_path,
    evaluation_strategy="no",  # Evaluate at the end of each epoch
    learning_rate=args.learning_rate,
    per_device_train_batch_size=args.per_device_train_batch_size,
    per_device_eval_batch_size=args.per_device_eval_batch_size,
    gradient_accumulation_steps=args.gradient_accumulation_steps,
    num_train_epochs=1,
    weight_decay=0.01,
    logging_dir="./logs",
    logging_steps=10,
    save_strategy="epoch",
    fp16=True,  # Enable mixed precision for better performance on supported hardware
    # bf16=True,
    report_to="none",  # Set to "wandb" if you are using Weights and Biases for logging
    dataloader_num_workers=4,
    # deepspeed="scripts/zero3.json",
    deepspeed="./scripts/zero3.json",
    ddp_find_unused_parameters=False
)

trainer = Trainer(
    model=model,
    args=training_args,
    train_dataset=tokenized_datasets['train'],
    # eval_dataset=tokenized_datasets['test'],  # Replace with a validation set if available
    data_collator=data_collator,
    tokenizer=tokenizer,
    preprocess_logits_for_metrics=preprocess_logits_for_metrics,
    compute_metrics=compute_metrics,
)

trainer.train()

--------------------------------

{'loss': 5.2096, 'grad_norm': 0.5192811814178179, 'learning_rate': 9.776785714285715e-06, 'epoch': 0.02}
{'loss': 5.0887, 'grad_norm': 0.34835128429484075, 'learning_rate': 9.575892857142857e-06, 'epoch': 0.04}
{'loss': 82.8303, 'grad_norm': 0.4701519288212055, 'learning_rate': 9.352678571428571e-06, 'epoch': 0.07}
{'loss': 11.6915, 'grad_norm': 0.44109291012967944, 'learning_rate': 9.151785714285715e-06, 'epoch': 0.09}
{'loss': 420.8393, 'grad_norm': 0.5913551485522904, 'learning_rate': 8.950892857142857e-06, 'epoch': 0.11}
...

I encountered unstable loss when using this code to fine tune the officially released math aps model.
Here "math-psa-7b" is openreasoner/Math-psa merged with Qwen2.5-Math-7B-Instruct

### Expected behavior

I want to know if there are any issues with my learning rate settings. 
And currently, the only dataset used is "openreasoner/MATH-APS". Can "prm800k_train.json" and "math-shepherd.json" be made public? The format of "math-shepherd.json" displayed on "peiyi9979/Math Shepherd" seems to be different from that of math-aps.json.
